core/Enclave/key_operation.h

```diff
@@ -0,0 +1,125 @@
+ */
+#include "log_utils.h"
+#include "sgx_tseal.h"
+
+#include <string>
+#include <stdio.h>
+#include <stdbool.h>
+#include <mbusafecrt.h>
+
+#include "sgx_report.h"
+#include "sgx_utils.h"
+#include "sgx_tkey_exchange.h"
+
+#ifndef _KEY_OPERATION_H_
+#define _KEY_OPERATION_H_
+
+void printf(const char *fmt, ...);
+
+sgx_status_t ehsm_aes_gcm_encrypt(const ehsm_data_t *aad,
+                                  const ehsm_keyblob_t *cmk,
+                                  const ehsm_data_t *plaintext,
+                                  ehsm_data_t *cipherblob);
+
+sgx_status_t ehsm_aes_gcm_decrypt(const ehsm_data_t *aad,
+                                  const ehsm_keyblob_t *cmk,
+                                  const ehsm_data_t *cipherblob,
+                                  ehsm_data_t *plaintext);
+
+sgx_status_t ehsm_sm4_ctr_encrypt(const ehsm_keyblob_t *cmk_blob,
+                                  const ehsm_data_t *plaintext,
+                                  ehsm_data_t *cipherblob);
+
+sgx_status_t ehsm_sm4_ctr_decrypt(const ehsm_keyblob_t *cmk_blob,
+                                  const ehsm_data_t *cipherblob,
+                                  ehsm_data_t *plaintext);
+
+sgx_status_t ehsm_sm4_cbc_encrypt(const ehsm_keyblob_t *cmk,
+                                  const ehsm_data_t *plaintext,
+                                  ehsm_data_t *cipherblob);
+
+sgx_status_t ehsm_sm4_cbc_decrypt(const ehsm_keyblob_t *cmk,
+                                  const ehsm_data_t *cipherblob,
+                                  ehsm_data_t *plaintext);
+
+sgx_status_t ehsm_rsa_encrypt(const ehsm_keyblob_t *cmk,
+                              ehsm_data_t *plaintext,
+                              ehsm_data_t *ciphertext);
+
+sgx_status_t ehsm_sm2_encrypt(const ehsm_keyblob_t *cmk,
+                              ehsm_data_t *plaintext,
+                              ehsm_data_t *ciphertext);
+
+sgx_status_t ehsm_rsa_decrypt(const ehsm_keyblob_t *cmk,
+                              ehsm_data_t *ciphertext,
+                              ehsm_data_t *plaintext);
+
+sgx_status_t ehsm_sm2_decrypt(const ehsm_keyblob_t *cmk,
+                              ehsm_data_t *ciphertext,
+                              ehsm_data_t *plaintext);
+
+sgx_status_t ehsm_rsa_sign(const ehsm_keyblob_t *cmk_blob,
+                           const ehsm_data_t *data,
+                           ehsm_data_t *signature);
+
+sgx_status_t ehsm_rsa_verify(const ehsm_keyblob_t *cmk,
+                             const ehsm_data_t *data,
+                             const ehsm_data_t *signature,
+                             bool *result);
+
+sgx_status_t ehsm_ec_encrypt(const ehsm_keyblob_t *cmk);
+
+sgx_status_t ehsm_ec_decrypt(const ehsm_keyblob_t *cmk);
+
+sgx_status_t ehsm_ecc_sign(const ehsm_keyblob_t *cmk,
+                           const ehsm_data_t *data,
+                           ehsm_data_t *signature,
+                           uint32_t *req_signature_len);
```

Comment:
why pass it? req_signature_len

Reply:
After ec sign, need to return an exact sign length used for ec verify